src/notification/NotificationController.ts

```diff
@@ -0,0 +1,266 @@
+import type { Patch } from 'immer';
+import { nanoid } from 'nanoid';
+
+import { hasProperty } from '../util';
+import { BaseController } from '../BaseControllerV2';
+
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+
+/**
+ * @type NotificationState
+ * @property notifications - Stores existing notifications to be shown in the UI
+ */
+export type NotificationState = {
+  notifications: Record<string, Notification>;
+};
+
+export type Notification = {
+  id: string;
+  type: NotificationType;
+  origin: string;
+  date: number;
```

Comment:
I had to look up where this was set to understand what date it referred to. Maybe `creationDate` or `createdDate` or something like that would be clearer?